index.js

```diff
@@ -330,7 +340,32 @@ function redisStore(args) {
       if (err) {
         return cb && cb(err);
       }
-      conn.keys(pattern, handleResponse(conn, cb));
+
+      // Use an object to dedupe as scan can return duplicates
+      var keysObj = {};
+      var count = 100;
+
+      (function nextBatch(cursorId) {
```

Comment:
Yeah, there's no easy way to do a `scan`. It has to be recursive, because the results are indeterminate. You have to keep calling it until redis returns a zero in the first element of the results. An IIFE is the easiest way to accomplish this.